src/block/block_write.c

```diff
@@ -21,6 +21,124 @@ __wt_block_header(WT_BLOCK *block)
 }
 
 /*
+ * __wt_block_truncate --
+ *	Truncate the file.
+ */
+int
+__wt_block_truncate(WT_SESSION_IMPL *session, WT_FH *fh, wt_off_t len)
+{
+	WT_RET(__wt_ftruncate(session, fh, len));
+
+	fh->size = fh->extend_size = len;
+
+	return (0);
+}
+
+/*
+ * __wt_block_extend --
+ *	Extend the file.
+ */
+static inline int
+__wt_block_extend(WT_SESSION_IMPL *session, WT_BLOCK *block,
+    WT_FH *fh, wt_off_t offset, size_t align_size, int *release_lockp)
+{
+	WT_DECL_RET;
+	int locked;
+
+	/*
+	 * The locking in this function is messy: the live system is locked when
+	 * we're called, by definition, but that lock may have been acquired by
+	 * our our caller or our caller's caller. If it's our caller's lock and
+	 * we can unlock it before returning (either before extending the file
+	 * or afterward, depending on the call used), then release_lock is set.
+	 *
+	 * If we unlock, but then find out we need a lock after all, re-acquire
+	 * the lock (and set release_lock so our caller knows to release it).
+	 */
+	locked = 1;
```

Comment:
Looking at the calling code, I know *release_lockp is set when called, but it seems like we should really do *release_lockp = 1 here just to pair it properly with the local knowledge that we're locked.  (Or maybe an assert.)